astropy/visualization/wcsaxes/core.py

```diff
@@ -489,7 +489,7 @@ def get_tightbbox(self, renderer):
         else:
             return self.get_window_extent(renderer)
 
-    def grid(self, b=None, axis='both', **kwargs):
+    def grid(self, b=None, axis='both', *, which='major', **kwargs):
```

Comment:
This diff doesn't look right.

Reply:
Without the `, *,` the `which` argument would become positional-or-keyword. To keep it keyword-only it needs the `*` "argument".

Reply:
Huh, I learned something new today! 🌈

Reply:
@MSeifert04 - why do this only in some cases though? And do we mind if it becomes a positional-or-keyword argument? (for this specific case I don't think I have a problem with it becoming positional-or-keyword)

Reply:
Just to clarify, I don't mind either way, I just want to understand the logic of when you decided to add ``*``?

Reply:
I tried to be conservative and not change the functionality in any way (except introspection, like `inspect.signature` called on any of the changed functions). So I added the `*` to all functions that had no `*args`. For those functions that had `*args` placing the arguments after the `*args` also makes them keyword-only. 

So if I have forgotten to use `*` in a function that didn't have `*args` it was an oversight.

Just in case I'm not good at explaining there's [PEP-3102](https://www.python.org/dev/peps/pep-3102/) which was responsible for the keyword-only-arguments in Python 3.